float64 x
float64 y
float64 z
float64 orw
float64 orx
float64 ory
float64 orz